cvat-core/src/frames.js

```diff
@@ -125,41 +169,56 @@
         });
     }
 
-    async function getFrame(taskID, mode, frame) {
+    async function getFrame(taskID, chunkSize, mode, frame) {
         if (!(taskID in frameDataCache)) {
-            frameDataCache[taskID] = {};
-            frameDataCache[taskID].meta = await serverProxy.frames.getMeta(taskID);
+            const blockType = mode === 'interpolation' ? cvatData.BlockType.TSVIDEO
+                : cvatData.BlockType.ARCHIVE;
+
+            const value = {
+                meta: await serverProxy.frames.getMeta(taskID),
+                chunkSize,
+                provider: new cvatData.FrameProvider(3, blockType, chunkSize),
+            };
 
             frameCache[taskID] = {};
+            frameDataCache[taskID] = value;
         }
-
-        if (!(frame in frameDataCache[taskID])) {
-            let size = null;
-            if (mode === 'interpolation') {
-                [size] = frameDataCache[taskID].meta;
-            } else if (mode === 'annotation') {
-                if (frame >= frameDataCache[taskID].meta.length) {
-                    throw new ArgumentError(
-                        `Meta information about frame ${frame} can't be received from the server`,
-                    );
-                } else {
-                    size = frameDataCache[taskID].meta[frame];
-                }
+        
+        let size = null;
+        if (mode === 'interpolation') {
+            [size] = frameDataCache[taskID].meta;            
+        } else if (mode === 'annotation') {
+            if (frame >= frameDataCache[taskID].meta.length) {
+                throw new ArgumentError(
+                    `Meta information about frame ${frame} can't be received from the server`,
+                );
             } else {
+                size = frameDataCache[taskID].meta[frame];
+            }
+        } else {
                 throw new ArgumentError(
                     `Invalid mode is specified ${mode}`,
                 );
             }
 
-            frameDataCache[taskID][frame] = new FrameData(size.width, size.height, taskID, frame);
```

Comment:
Maybe is it better to cache these instances like before? It isn't exactly frames. They are only wrappers which don't have binary information about frames.

Reply:
No!!! I realy need cache in cvat-data due to worker using.